### ISSUE TITLE
wiki.getAttachment fails if remoteuser has no permission to [root]

### ISSUE THREAD

**Issue by eduardomozart** [splitbrain/dokuwiki#3552]
Hello Andreas Gohr and DokuWiki community,

Thank you and all DokuWiki community for making this project possible. DokuWiki is amazing and I've been using it by some years now.

I would like to report what I believe is a bug report: if user has no permission to "[root]", the ``wiki.getAttachments`` XML-RPC call fails with a ``transport error - HTTP status 403`` error, instead of returning the list of media attachments that the group specified at 'remoteuser' parameter has access to.

It doesn't appears to be related to SSL connection issues, because both the wikis are using a non-expired Let's Encrypt certificate, both the wikis are hosted into the same webserver and the ``dokuwiki.getPagelist`` XML-RPC call works as expected, returning a list of pages that the group specified at 'remoteuser' parameter has access to.

I opened a bug report at https://github.com/splitbrain/dokuwiki-plugin-sync/issues/56 about this issue and proposed a dirty fix for the Sync plugin at https://github.com/splitbrain/dokuwiki-plugin-sync/pull/57, but as I believe it maybe a DokuWiki missbehavior I decided to open this bug report.

**Comment by Klap-in** [comment]
Could you give an example of file structure and permission structure?

Let's me guess:
ACL
```
*                  @ALL           0
namespace:*                  @remoteusergroep          16
```
files
```
/media1.png   (not accessible)
/namespace:media2.png (accessible)
/namespace:subnamespace:media3.png (accessible)
```

Could you tell which parameters you set for the `dokuwiki.getAttachments` call?
Which call do you use for the `dokuwiki.getPageList`?

background:

Source code of the list function:
https://github.com/splitbrain/dokuwiki/blob/40bf970f9b3c190fc0bfc3e11605ade618450440/inc/Remote/ApiCore.php#L425-L452

Existing unit tests:
https://github.com/splitbrain/dokuwiki/blob/40bf970f9b3c190fc0bfc3e11605ade618450440/_test/tests/inc/remoteapicore.test.php#L584-L609

**Comment by eduardomozart** [comment]
Hello @Klap-in, I'm using exactly the file structure and permission structure you provided. The ``wiki.getAttachments`` call that I'm using is from Sync DokuWiki's plugin - https://github.com/splitbrain/dokuwiki-plugin-sync/blob/master/admin.php#L656

**Comment by eduardomozart** [comment]
No, depth is empty (Sync plugin is using "0" as value for depth) and namespace is "" (empty). It works if I manually specifies a namespace that the user group has access to.
